sys/clif/clif.c

```diff
@@ -278,7 +278,9 @@ ssize_t clif_get_attr(const char *input, size_t input_len, clif_attr_t *attr)
             attr->key_len = pos - attr->key;
             /* check if the value is quoted and prepare pointer for value scan */
             pos++;
-            if (*pos == '"') {
+            if (pos == end)
+                break;
```

Comment:
If it gets to this condition, it would be a malformed attribute  right? (e.g. `foo=`)
```suggestion
                return CLIF_NOT_FOUND;
```

Reply:
Right, but what about `foo="`? That doesn't seem to return `CLIF_NOT_FOUND` either at the moment or am I missing something here?

Reply:
Yeah, we should check for that, and also a possible `foo="ba` I guess, and return `CLIF_NOT_FOUND`

Reply:
Ok, maybe it makes sense to implement this separately? The changes I propose here handle `foo=` in the same way as `foo="` is handled presently.

Reply:
Ok. Could you add your proposed test to the unit test application?

Reply:
Done, keep in mind though that the tests are not run with ASAN enabled. So even if this bug would be reintroduced they would not necessarily detected it.